src/NeoSharp.Core/Persistence/IDbPersistenceRepository.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace NeoSharp.Core.Persistence
+{
+    public interface IDbPersistenceRepository
+    {
+        IBlockHeaderRepository BlockHeaderRepository { get; }
+
+        ITransactionRepository TransactionRepository { get; }
```

Comment:
Please do not create kinda global object, favor explicit dependencies instead.

Reply:
therefore we will have several injections in classes like "Blockchain" that need this .. are you sure?

Reply:
I think it is much better than everyone will always inject IDbPersistenceRepository and use specific repos through the properties. On the side note, Blockchain object is quite big and I like to tear this apart to single responsibility classes.

Reply:
ok .. got it!! will implement like that!